src/main/kotlin/sliv/tool/parsers/planes/ImagePlanesParser.kt

```diff
@@ -0,0 +1,108 @@
+package sliv.tool.parsers.planes
+
+import java.awt.image.DataBufferByte
+import kotlin.io.path.Path
+import kotlin.io.path.extension
+import sliv.tool.parsers.Parser
+import sliv.tool.parsers.ParserUtils
+import sliv.tool.parsers.structures.Plane
+import sliv.tool.scene.model.Point
+import java.awt.image.BufferedImage
+
+// Describes types of image formats, according to their color segments number.
+private enum class ColorSegmentsType {
+    TRIPLE {
+        override val colorComponentsNumber = 3
+        override val segmentsByteOffset = 0
+    },
+    QUAD {
+        override val colorComponentsNumber = 4
+        override val segmentsByteOffset = 1
+    };
+
+    abstract val colorComponentsNumber: Int
+    abstract val segmentsByteOffset: Int
+}
+
+// A parser class for planes stored in images in a form of a mask.
+class ImagePlanesParser(private val colorToUIDMap: Map<Int, Long>) : Parser<Plane> {
+    companion object {
+        private const val ColorsPerSegment = 256
+    }
+
+    private fun convertSeparateToWholeRGB(r: Int, g: Int, b: Int): Int =
+        (r * ColorsPerSegment * ColorsPerSegment + g * ColorsPerSegment + b)
+
+    private fun getPlanePixelColor(
+        pixelIndex: Int,
+        imageByteDataArray: ByteArray,
+        segmentsType: ColorSegmentsType
+    ): Int {
+        val segmentsByteOffset = segmentsType.segmentsByteOffset
+
+        return convertSeparateToWholeRGB(
```

Comment:
As I understood, no. We need to convert byte to ubyte, and then to int, since the highest bit is a sign bit by default. And there are no methods in Kotlin to convert bits to int.